app/Services/UserService.php

```diff
@@ -63,4 +66,23 @@ public function passwordResetLink(array $validatedForgetPassword): string
     {
         return Password::sendResetLink($validatedForgetPassword);
     }
+
+    public function updatePassword(array $validatedResetPassword): mixed
+    {
+        $user = User::where('email', $validatedResetPassword['email'])->first();
+
+        if (Hash::check($validatedResetPassword['password'], $user->password)) {
+            return throw new BadRequestException('cannot use old password as a password reset');
+        }
+
+        return Password::reset($validatedResetPassword, function ($user, $password) {
+            $user->forceFill(
+                ['password' => $password]
+            )->setRememberToken(Str::random(60));
+
+            $user->tokens()->delete();
```

Comment:
@sabinshrestha8 Any reason for deleting tokens?

Reply:
Since our token doesn't have expiry date and I came to realization that after resetting the password the devices who are already logged in and haven't logout, it can proceed its activity of work without any interruption even after password reset so, for the security purpose, I thought of deleting tokens is must.